File: BSBackEnd/Models/Pedido.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace BSBackEnd.Models
{
    public class Pedido
    {
        public Guid Id { get; set; }
        [Required]
        public Guid CompradorId { get; set; }        //Usuário que está realizando o pedido de compra             
        public Usuario Comprador { get; set; }       //Usuário que está realizando o pedido de compra        
        [Required]
        public Guid AnuncioId { get; set; }          //Anuncio escolhido    
        public Anuncio Anuncio { get; set; }         //Anuncio escolhido    
        [Required(ErrorMessage = "Quantidade obrigatória")]
        public int Qtde { get; set; }                //Quantidade solicitada (se existir em estoque)
        [Required(ErrorMessage = "Informe se deseja entrega")]
        public bool SolicitaEntrega { get; set; }    //Se o comprador deseja que entreguem o produto (caso o vendedor forneça entrega)
        [Required]
        public DateTime DataPedido { get; set; }
    }
}

File: bsmobile/lib/models/Pedido.dart
import 'dart:convert';
import 'package:intl/intl.dart';

import 'Anuncio.dart';


class Pedido {
  String id;
  String compradorId;
  //public Usuario Comprador { get; set; }
  String anuncioId;
  Anuncio anuncio;
  int qtde;
  bool solicitaEntrega;
  DateTime dataHoraPedido;
  String dataPedido;
  

  Pedido(this.id, this.compradorId, this.anuncioId, this.anuncio, this.qtde,
      this.solicitaEntrega, this.dataPedido);

  Pedido.fromJson(Map<String, dynamic> json) {
    this.id = json['id'];
    this.compradorId = json['compradorId'];
    this.anuncioId = json['anuncioId'];
    this.qtde = json['qtde'];
    this.solicitaEntrega = json['solicitaEntrega'];
    this.dataHoraPedido = DateTime.parse(json['dataPedido']);
    this.dataPedido =  new DateFormat("dd/MM/yyyy").format(dataHoraPedido);

    this.anuncio = Anuncio.fromJson(json['anuncio']);  
  }
}
